android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java

```diff
@@ -36,26 +39,23 @@
 
 public class AblyMethodCallHandler implements MethodChannel.MethodCallHandler {
   private static AblyMethodCallHandler _instance;
+  private Context applicationContext;
 
   public interface OnHotRestart {
     // this method will be called on every fresh start and on every hot restart
     void trigger();
   }
 
-  static synchronized AblyMethodCallHandler getInstance(final MethodChannel channel, final OnHotRestart listener, final Context applicationContext) {
-    if (null == _instance) {
-      _instance = new AblyMethodCallHandler(channel, listener, applicationContext);
-    }
-    return _instance;
-  }
-
   private final MethodChannel channel;
   private final OnHotRestart onHotRestartListener;
   private final Map<String, BiConsumer<MethodCall, MethodChannel.Result>> _map;
```

Comment:
_map is a method of method strings to implementations. If the java side receives this string when method is invoked from the Dart side, the implementation is called. 

The reason why its `_map` (quite short), is because it used very often, where each line is also quite long, e.g.: `_map.put(PlatformConstants.PlatformMethod.getPlatformVersion, this::getPlatformVersion);`. 

It's also possible to remove the map completely, and just use a switch statement instead of having map accesses. This is many plugins implement it. It looks much clearer, but the lines will be double as long (because switch statements take up more space 😅).

WDYT, should it be short or longer?

Reply:
https://github.com/ably/ably-flutter/issues/180